luigi/tasks/genome_mapping/genome_mapping_tasks.py

```diff
@@ -0,0 +1,395 @@
+# -*- coding: utf-8 -*-
+
+"""
+Copyright [2009-2018] EMBL-European Bioinformatics Institute
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import os
+import json
+import subprocess
+
+import luigi
+import requests
+
+from glob import iglob
+
+from tasks.config import db
+from tasks.config import genome_mapping
+
+from tasks.export.ftp.fasta.utils import FastaExportBase
+
+from rnacentral.genome_mapping import genome_mapping as gm
+from rnacentral.psql import PsqlWrapper
+
+
+def get_mapped_assemblies():
+    """
+    Get metadata about all assemblies.
+    """
+    psql = PsqlWrapper(db())
+    sql = """
+    select assembly_id, assembly_ucsc, taxid, ensembl_url as species,
+           division, subdomain
+    from ensembl_assembly
+    """
+    assemblies = []
+    for assembly in psql.copy_to_iterable(sql):
+        assemblies.append(assembly)
+        assemblies[-1]['taxid'] = int(assemblies[-1]['taxid'])
+    return assemblies
+
+
+class GetFasta(FastaExportBase):
+    """
+    Export RNAcentral sequences for a particular species to a FASTA file.
+    """
+    taxid = luigi.IntParameter()
+    species = luigi.Parameter()
+
+    def output(self):
+        filename = os.path.join(genome_mapping().rnacentral_fasta(self.species), 'rnacentral.fa')
+        return luigi.LocalTarget(filename)
+
+    def records(self):
+        return gm.export_rnacentral_fasta(db(), taxid=self.taxid)
+
+
+class CleanSplitFasta(luigi.Task):
+    """
+    Use seqkit to split RNAcentral species-specific fasta files into chunks to
+    speed up blat searches. Filter out sequences that are too short or too long
+    to be mapped with blat.
+    """
+    taxid = luigi.IntParameter()
+    species = luigi.Parameter()
+    num_seq = luigi.IntParameter(default=1000)
+    min_length = luigi.IntParameter(default=20)
+    max_length = luigi.IntParameter(default=100000)
+    prefix = 'rnacentral-clean.part_'
+
+    def requires(self):
+        return GetFasta(taxid=self.taxid, species=self.species)
+
+    def output(self):
+        chunk_fasta = os.path.join(genome_mapping().chunks(self.species),
+                                   '{}*.fasta'.format(self.prefix))
+        fasta_files = []
+        for filename in iglob(chunk_fasta):
+            fasta_files.append(luigi.LocalTarget(filename))
+        return fasta_files
+
+    def run(self):
+        cmd = ('seqkit seq --min-len {min_length} --max-len {max_length} '
```

Comment:
I've started to use [plumbum](https://pypi.org/project/plumbum/) to deal with more complex shell commands. It is recommended in the Luigi documentation as well.

Reply:
Whaaat! this is an awesome library, will use it in the future.
https://pypi.org/project/plumbum/
One could argue that plumbum makes the code harder to debug outside of python but I can see how it can be useful too.